src/NLog.Targets.Syslog/Policies/ParamNamePolicySet.cs

```diff
@@ -28,9 +27,10 @@ public ParamNamePolicySet(EnforcementConfig enforcementConfig)
 
         public string Apply(string s, string searchFor)
         {
-            var afterBasicPolicies = basicPolicies
-                .Where(p => p.IsApplicable())
-                .Aggregate(s, (acc, curr) => curr.Apply(acc));
+            var afterBasicPolicies = s;
+            foreach (var policy in basicPolicies)
+                if (policy.IsApplicable())
+                    afterBasicPolicies = policy.Apply(afterBasicPolicies);
 
```

Comment:
Is there an allocation advantage to remove LINQ?

Reply:
```
var afterBasicPolicies = basicPolicies
            .Where(p => p.IsApplicable()) // Allocates WhereEnumerableIterator
            .Aggregate(s, (acc, curr) => curr.Apply(acc)); // Allocates IEnumerator
```
The updated code only allocates when policy transformation is needed.